react/features/overlay/components/OverlayContainer.js

```diff
@@ -0,0 +1,212 @@
+/* global APP */
+import React, { Component } from 'react';
+import { connect } from 'react-redux';
+
+import PageReloadOverlay from './PageReloadOverlay';
+import SuspendedOverlay from './SuspendedOverlay';
+import UserMediaPermissionsOverlay from './UserMediaPermissionsOverlay';
+
+/**
+ * Implements a React Component that will display the correct overlay when
+ * needed.
+ */
+class OverlayContainer extends Component {
+    /**
+     * OverlayContainer component's property types.
+     *
+     * @static
+     */
+    static propTypes = {
```

Comment:
Where do you think the decision about the value of `overlayType` should be made? The reducer maybe?